commit cf988760424fd5b3c6eced4f58e2398765278808 (refs/changes/29/4529/3)
Author: Dan McDonald <danmcd@joyent.com>
Date:   2018-07-17T18:59:58-04:00 (1 year, 3 months ago)
    
    OS-7081 illumos-extra dmake clobbers errno in failure cases
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Reviewed by: Jason King <jbk@joyent.com>
    Approved by: Jason King <jbk@joyent.com>

diff --git a/make/dmake.patch b/make/dmake.patch
index c66ee7c..13e9479 100644
--- a/make/dmake.patch
+++ b/make/dmake.patch
@@ -309,6 +309,30 @@ index fc4306b..1211caf 100644
  #if defined(DISTRIBUTED)
  #	include <dm/Avo_CmdOutput.h>
  #	include <rw/xdrstrea.h>
+@@ -401,8 +403,9 @@
+ 		}
+ 	}
+ 	if (childPid  == -1) {
+-		fatal_mksh(catgets(libmksdmsi18n_catd, 1, 94, "fork failed: %s"),
+-		      errmsg(errno));
++		int save_errno = errno;
++		fatal_mksh(catgets(libmksdmsi18n_catd, 1, 94, "doshell() fork failed: %s"),
++		      errmsg(save_errno));
+ 	}
+ 	retmem_mb(argv[cmd_argv_index]);
+ 	return childPid;
+@@ -591,8 +594,9 @@
+ 		fatal_mksh(catgets(libmksdmsi18n_catd, 1, 96, "Cannot load command `%s': %s"), argv[1], errmsg(errno));
+ 	}
+ 	if (childPid  == -1) {
+-		fatal_mksh(catgets(libmksdmsi18n_catd, 1, 97, "fork failed: %s"),
+-		      errmsg(errno));
++		int save_errno = errno;
++		fatal_mksh(catgets(libmksdmsi18n_catd, 1, 97, "doexec() fork failed: %s"),
++		      errmsg(save_errno));
+ 	}
+ 	for (int i = 0; argv[i] != NULL; i++) {
+ 		retmem_mb(argv[i]);
 diff --git a/src/make_src/Make/lib/mksh/src/mksh.cc b/src/make_src/Make/lib/mksh/src/mksh.cc
 index 98c9601..8004ac2 100644
 --- a/src/make_src/Make/lib/mksh/src/mksh.cc
